### ISSUE TITLE
FL Studio Installation Failed (Installer is not well configured.)

### ISSUE THREAD

**Issue by maureenferreira** [bottlesdevs/programs#337]

**Comment by tiddymasher** [comment]
I The .exe is outdated. I sincerely doubt this version of FL studio works with VST plugins either, though. Bottles runs FL the best, but its still very unstable. I was hoping to use the pre-configured installer even though the version of FL is slightly out of date but I also ran into the same issue.

**Comment by tyjman** [comment]
I was able to manually get FL Studio working in Bottles 

1. Create a new bottle
2. Install the "allfonts" dependency
3. Disable DXVK
4. Run the FL Studio Installer
5. After a while the installer will hang on a "WebView" thing, you can just kill this process and the installer will finish

Now FL Studio should launch, but FLEX crashes, I got it working again by re-enabling DXVK.
Then I had problems with fabfilter, but I solved that by installing vcrun6 (which is called vcredist6 in bottles)

**Comment by David-Everlasting-Media** [comment]
> I was able to manually get FL Studio working in Bottles
> 
> 1. Create a new bottle
> 2. Install the "allfonts" dependency
> 3. Disable DXVK
> 4. Run the FL Studio installer (doesn't render properly with DXVK enabled for some reason)
> 5. After a while the installer will hang on a "WebView" thing, you can just kill the "WebView" process and the installer will finish
> 
> Now FL Studio should launch, but FLEX crashes, I got it working again by re-enabling DXVK. Then I had problems with fabfilter, but I solved that by installing vcrun6 (which is called vcredist6 in bottles)

I was unable to install it. "Installer poorly configured" or whatever. RIP.

**Comment by tyjman** [comment]
@David-Everlasting-Media 

> I was unable to install it. "Installer poorly configured" or whatever. RIP.

Sounds like you're running the program installer (going to the bottle -> Install Programs... -> and then choosing FL Studio in that list) but what I meant when I said "FL Studio installer" is not that installer but the actual FL Studio executable installer that you manually download.

**Comment by khlobbia** [comment]
> I was able to manually get FL Studio working in Bottles
> 
>     1. Create a new bottle
> 
>     2. Install the "allfonts" dependency
> 
>     3. Disable DXVK
> 
>     4. Run the FL Studio installer (doesn't render properly with DXVK enabled for some reason)
> 
>     5. After a while the installer will hang on a "WebView" thing, you can just kill the "WebView" process and the installer will finish
> 
> 
> Now FL Studio should launch, but FLEX crashes, I got it working again by re-enabling DXVK. Then I had problems with fabfilter, but I solved that by installing vcrun6 (which is called vcredist6 in bottles)

Hi, got that working and finished installing it, but I don't know how to launch FL. I double-checked the installation location 
(C:\Program FIles\Image-Line\FL Studio 2024\) But going there with my file manager I can't find it. Maybe it's an issue with where I downloaded the installer?

**Comment by tyjman** [comment]
> Hi, got that working and finished installing it, but I don't know how to launch FL. I double-checked the installation location (C:\Program FIles\Image-Line\FL Studio 2024) But going there with my file manager I can't find it. Maybe it's an issue with where I downloaded the installer?

Well, when mine finished installing it actually automatically made a shortcut on my desktop, but it also popped up inside the bottle (1) which can be ran with the play button (2):

![2024-11-26_20-59](https://github.com/user-attachments/assets/bcf1d029-9722-4706-adb4-775b491870ab)

Remember that Bottles has its own paths, I don't know where you went with your file manager, so your safest bet to actually check the files is by hitting the kebab menu (3), meaning the three vertical dots, and clicking "browse files" from the menu that drops down, then you can see if the files are there. If they're not then I guess it didn't install properly, try again.

**Comment by CaitlynWemyss** [comment]
At least on my machine, this issue is due to an HTTP 403 error (forbidden) when attempting to download the installer.
![image](https://github.com/user-attachments/assets/9817334c-29ea-4421-9b38-74cbd2ebb58b)
Wget also throws the same error when attempting to download from that link.
![image](https://github.com/user-attachments/assets/c1a8c934-cf14-491f-98cb-5a33a0bc1ac1)
However, opening the link in a browser downloads it correctly. I don't know what exactly is causing this behaviour, but it seems to be some security measure with FL Studio's download servers.

**Comment by tyjman** [comment]
It might be forbidden, it might just be removed. The latest version works: https://install.image-line.com/flstudio/flstudio_win64_24.2.1.4526.exe

A long term solution would probably be to use https://support.image-line.com/redirect/flstudio_win_installer instead since that redirects to the latest exe.

**Comment by CaitlynWemyss** [comment]
The current link (24.1.1.4239) is still a working link *if you download it from a browser.* The redirect, if downloaded with Wget, successfully redirects to the current version (24.2.1.4526) but then also throws a 403.
![image](https://github.com/user-attachments/assets/bd673ee6-2968-47f7-a971-5f2f3b4c64a8)

The failure to download using Wget and the success when downloading with a browser is consistent across all links I've tried.

**Comment by Unicornssss7428** [comment]
@CaitlynWemyss 

Using DevTools and the copy as PowerShell command, one can see the headers required for a 200 response to the request.

Invoke-WebRequest -UseBasicParsing -Uri "https://support.image-line.com/redirect/fl_studio_win_installer" `
-UserAgent "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:134.0) Gecko/20100101 Firefox/134.0" `
-Headers @{
"Accept" = "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"
  "Accept-Language" = "en-US,en;q=0.5"
  "Accept-Encoding" = "gzip, deflate, br, zstd"
  "DNT" = "1"
  "Sec-GPC" = "1"
  "Upgrade-Insecure-Requests" = "1"
  "Sec-Fetch-Dest" = "document"
  "Sec-Fetch-Mode" = "navigate"
  "Sec-Fetch-Site" = "none"
  "Sec-Fetch-User" = "?1"
  "Priority" = "u=0, i"
  "TE" = "trailers"
}

For POSIX shells:


curl 'https://support.image-line.com/redirect/fl_studio_win_installer' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:134.0) Gecko/20100101 Firefox/134.0' -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' -H 'Accept-Language: en-US,en;q=0.5' -H 'Accept-Encoding: gzip, deflate, br, zstd' -H 'DNT: 1' -H 'Sec-GPC: 1' -H 'Connection: keep-alive' -H 'Upgrade-Insecure-Requests: 1' -H 'Sec-Fetch-Dest: document' -H 'Sec-Fetch-Mode: navigate' -H 'Sec-Fetch-Site: none' -H 'Sec-Fetch-User: ?1' -H 'Priority: u=0, i' -H 'TE: trailers'



Using only the user agent seems not to work, so one of these other headers is fixing something. Not sure exactly which one.

**Comment by CaitlynWemyss** [comment]
@Unicornssss7428 Thanks for the tip!

Unfortunately, after some experimenting trying to figure out the solution by process of elimination, I think I ended up getting locked out.

Me running the exact same command twice in a row:

```
wget -U "Mozilla/5.0 (X11; Linux x86_64; rv:133.0) Gecko/20100101 Firefox/134.0" --header='Accept-Language: en-US,en;q=0.5' https://install.image-line.com/flstudio/flstudio_win64_24.2.1.4526.exe
--2025-01-22 11:16:02--  https://install.image-line.com/flstudio/flstudio_win64_24.2.1.4526.exe
Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving install.image-line.com (install.image-line.com)... 104.16.96.54, 104.16.95.54
Connecting to install.image-line.com (install.image-line.com)|104.16.96.54|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1078066888 (1.0G) [application/octet-stream]
Saving to: ‘flstudio_win64_24.2.1.4526.exe’

          flstudio_  10%[=>                  ] 105.40M  3.65MB/s    eta 4m 24s
```

```
wget -U "Mozilla/5.0 (X11; Linux x86_64; rv:133.0) Gecko/20100101 Firefox/134.0" --header='Accept-Language: en-US,en;q=0.5' https://install.image-line.com/flstudio/flstudio_win64_24.2.1.4526.exe
--2025-01-22 11:19:11--  https://install.image-line.com/flstudio/flstudio_win64_24.2.1.4526.exe
Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving install.image-line.com (install.image-line.com)... 104.16.96.54, 104.16.95.54
Connecting to install.image-line.com (install.image-line.com)|104.16.96.54|:443... connected.
HTTP request sent, awaiting response... 403 Forbidden
2025-01-22 11:19:11 ERROR 403: Forbidden.
```

Now that I'm locked out even using all the headers you provided I still can't access the file. Before I got locked out though, I determined the answer was an up to date user agent plus the Accept-Language header.

I've been banging my head against the wall over this problem so much at this point and I'm starting to think Image Line just really don't want people automatically downloading their files, most likely to prevent DDOS attacks or something. If they have an adaptive system that locks people out after they detect suspicious activity, it may not be possible to resolve this issue at all.

**Comment by Unicornssss7428** [comment]
@CaitlynWemyss From looking at their cookies I noticed a cf_clearance cookie, which seems to be some sort of CloudFlare related challenge cookie. So your theory about it preventing DDOS attacks is likely correct, CloudFlare probably internally looks through the http headers/amount of requests/cookies etc. to determine whether or not to send you a 403, but I'm just speculating at it's internals. 

So, with your "may not possible to resolve this issue at all" I think you're right unless we find a cloudflare bypass, which would be characterized as a vulnerability in cloudflare and promptly fixed haha.

What Image-Line should do is create a small API handler so we can download files without the need for the browser, but I *highly* doubt they have a large enough user base that requires something like this so this is likely a fever dream.

One way this could be "fixed" is having a mirror somewhere, but then that presents the issue of trusting that server's files, which with the proper checksum/hash/signature (I alwasys confuse these words) could work fine.


Someone dedicates a VPS to hosting the FL studio installer download for bottles/whatever needs it to request it from, bottles checks the file to make sure it matches the last known hash of the latest file, if it's a match it's considered good and it downloads.


This solution would also require someone from the community to pay for hosting the file, then the bottles devs to actually give a damn about it. As far as I remember, the main dev of bottles is focusing on a different project at the moment per their discord.

Sorry for the long rant, 

TL;DR Yeah, I agree it's DDOS protection that makes this a non-bottles issue and unfixable.

**Comment by Unicornssss7428** [comment]
The other "solution", which doesn't solve the "not able to download using curl/wget" issue, is to download the exe manually from the site and start it within bottles.

At least you can actually get it installed which is the end goal afterall,

#362 

From DaruMonkey:

```
In the meantime, you can download the installer from the Image-Line website and run it from the bottle created for FL Studio.

The installer hung on the final step (installing MS Edge bits). I used the bottle's task manager to kill the process, and the install completed.

FL Studio 24 works, although I haven't tested any VSTs yet.
```

**Comment by Axis4s** [comment]
I Think of instead of downloading the installer for you it should ask the user to provide the executable.

**Comment by geekeie** [comment]
### ✅ Still Having Trouble?
If you’re getting the error:

FL Studio Installation Failed (Installer is not well configured.) #337

It usually means the installer is corrupted or incomplete.

**💡 Solution:**
Download the official and correctly configured installer from Image-Line:[ image-line.com](https://image-line.com)

If you want free version on PC then [download fl studio crack version
](https://flstudiox.com)
This free version gives you full access to all features, and it avoids issues like #337 caused by broken or modified installers.
